six.py

```diff
@@ -818,6 +818,11 @@ def wrapper(f):
 
 def with_metaclass(meta, *bases):
     """Create a base class with a metaclass."""
+    # No need for a dummy metaclass if `meta` is already present in
+    # the inheritance hierarchy. See also a note in `add_metaclass`.
+    if any(isinstance(cls, meta) for cls in bases):
+        return type('temporary_class', bases, {})
```

Comment:
changing this to `meta = type` fixes the case handled at head where an ancestor in `bases` (e.g. `typing.Generic[T]`) needs `__bases__` to be retained.